### PR TITLE
Unify clinicalDataEqualityFilters and clinicalDataIntervalFilters into clinicalDataFilters

### PR DESCRIPTION
**Pull request by kalletlak**
Fix part 3 of https://github.com/cBioPortal/cbioportal/issues/4983#issuecomment-454933667

### CODE REVIEW

File: src/pages/groupComparison/GroupComparisonUtils.tsx [hunk 1]
```diff
@@ -389,12 +389,15 @@ export function CLINICAL_TAB_NOT_ENOUGH_GROUPS_MSG(numSelectedGroups:number) {
...
+    customChartFilterSet:{[chartId:string]:string[]},
+    clinicalAttributeIdToDataType:{[chartId:string]:string}
 ) {
-    const equalityFilters = _.sortBy( // sort clinical data equality filters into a canonical order - lets just do alphabetical by attribute id
-        filters.clinicalDataEqualityFilters || [],
-        filter=>filter.attributeId
-    ).map(filter=>filter.values.join("+")); // get each attributes selected values, joined by +
+
+    const equalityFilters = _.chain(filters.clinicalDataFilters || [])
+        .filter(clinicalDataFilter => clinicalAttributeIdToDataType[clinicalDataFilter.attributeId] === "STRING")
```
Comment (zhx828) [inline src/pages/groupComparison/GroupComparisonUtils.tsx]: Let's put the STRING to a constant

File: src/pages/studyView/UserSelections.tsx [hunk 2]
```diff
@@ -93,47 +93,44 @@ export default class UserSelections extends React.Component<IUserSelectionsProps
...
+                                    <GroupLogic components={clinicalDataFilter.values.map(clinicalDataFilterValue => {
+                                        return <PillTag
+                                            content={clinicalDataFilterValue.value}
+                                            backgroundColor={STUDY_VIEW_CONFIG.colors.theme.clinicalFilterContent}
+                                            onDelete={() => this.props.updateClinicalDataFilterByValues(chartMeta, _.remove(clinicalDataFilter.values, value => value.value !== clinicalDataFilterValue.value))}
+                                        />
+                                    })} operation={'or'} group={false}/>
+                                ]}
+                                operation={':'}
+                                group={false}/></div>
```
Comment (onursumer) [inline src/pages/studyView/UserSelections.tsx]:
```suggestion
                                group={false}/>
                            </div>
```

File: src/pages/studyView/UserSelections.tsx [hunk 3]
```diff
@@ -93,47 +93,44 @@ export default class UserSelections extends React.Component<IUserSelectionsProps
...
+                    acc.push(
+                        <div className={styles.parentGroupLogic}>
+                            <GroupLogic
+                                components={[
+                                    <span className={styles.filterClinicalAttrName}>{chartMeta.displayName}</span>,
+                                    <GroupLogic components={clinicalDataFilter.values.map(clinicalDataFilterValue => {
+                                        return <PillTag
+                                            content={clinicalDataFilterValue.value}
+                                            backgroundColor={STUDY_VIEW_CONFIG.colors.theme.clinicalFilterContent}
+                                            onDelete={() => this.props.updateClinicalDataFilterByValues(chartMeta, _.remove(clinicalDataFilter.values, value => value.value !== clinicalDataFilterValue.value))}
```
Comment (onursumer) [inline src/pages/studyView/UserSelections.tsx]:
```suggestion
                                            onDelete={() => this.props.updateClinicalDataFilterByValues(chartMeta,
                                                _.remove(clinicalDataFilter.values, value => value.value !== clinicalDataFilterValue.value))}
```

File: src/pages/studyView/UserSelections.tsx [hunk 4]
```diff
@@ -93,47 +93,44 @@ export default class UserSelections extends React.Component<IUserSelectionsProps
...
-                                    />
-                                })} operation={'or'} group={false}/>
+                                <PillTag
+                                    content={intervalFiltersDisplayValue(clinicalDataFilter.values)}
+                                    backgroundColor={STUDY_VIEW_CONFIG.colors.theme.clinicalFilterContent}
+                                    onDelete={() => this.props.updateClinicalDataFilterByValues(chartMeta, [])}
+                                />
                             ]}
                             operation={':'}
                             group={false}/></div>
```
Comment (onursumer) [inline src/pages/studyView/UserSelections.tsx]:
```suggestion
                             group={false}/>
                         </div>
```

File: src/pages/studyView/UserSelections.tsx [hunk 5]
```diff
@@ -93,47 +93,44 @@ export default class UserSelections extends React.Component<IUserSelectionsProps
...
-        _.reduce((this.props.filter.clinicalDataEqualityFilters || []), (acc, clinicalDataEqualityFilter) => {
-            const chartMeta = this.props.attributesMetaSet[clinicalDataEqualityFilter.attributeId];
+        _.reduce((this.props.filter.clinicalDataFilters || []), (acc, clinicalDataFilter) => {
+            const chartMeta = this.props.attributesMetaSet[clinicalDataFilter.attributeId];
             if (chartMeta) {
-                acc.push(
-                    <div className={styles.parentGroupLogic}>
-                        <GroupLogic
+                const dataType = this.props.clinicalAttributeIdToDataType[clinicalDataFilter.attributeId];
+                if(dataType === "STRING") {
```
Comment (onursumer) [inline src/pages/studyView/UserSelections.tsx]: add a "STRING" constant as @zhx828 suggested above.

**Comment by onursumer** [comment]
@kalletlak Seems like `install` is failing due to this TS error:
```
[tsl] ERROR in /tmp/repo/src/pages/studyView/tabs/ClinicalDataTab.tsx(82,185)
      TS2304: Cannot find name 'DataType'
```

File: src/pages/studyView/UserSelections.tsx [hunk 6]
```diff
@@ -93,47 +93,48 @@ export default class UserSelections extends React.Component<IUserSelectionsProps
...
+                                        return <PillTag
+                                            content={clinicalDataFilterValue.value}
+                                            backgroundColor={STUDY_VIEW_CONFIG.colors.theme.clinicalFilterContent}
+                                            onDelete={() => this.props.updateClinicalDataFilterByValues(chartMeta,
+                                                _.remove(clinicalDataFilter.values, value => value.value !== clinicalDataFilterValue.value))}
+                                        />
+                                    })} operation={'or'} group={false} />
+                                ]}
+                                operation={':'}
+                                group={false} />
```
Comment (onursumer) [inline src/pages/studyView/UserSelections.tsx]:
```suggestion
                                group={false}
                            />
```

File: src/pages/studyView/UserSelections.tsx [hunk 7]
```diff
@@ -93,47 +93,48 @@ export default class UserSelections extends React.Component<IUserSelectionsProps
...
-                                        content={label}
+                const dataType = this.props.clinicalAttributeIdToDataType[clinicalDataFilter.attributeId];
+                if(dataType === DataType.STRING) {
+                    // Pie chart filter
+                    acc.push(
+                        <div className={styles.parentGroupLogic}>
+                            <GroupLogic
+                                components={[
+                                    <span className={styles.filterClinicalAttrName}>{chartMeta.displayName}</span>,
+                                    <GroupLogic components={clinicalDataFilter.values.map(clinicalDataFilterValue => {
```
Comment (onursumer) [inline src/pages/studyView/UserSelections.tsx]:
```suggestion
                                    <GroupLogic 
                                        components={clinicalDataFilter.values.map(clinicalDataFilterValue => {
```

File: src/pages/studyView/UserSelections.tsx [hunk 8]
```diff
@@ -93,47 +93,48 @@ export default class UserSelections extends React.Component<IUserSelectionsProps
...
+                                        return <PillTag
+                                            content={clinicalDataFilterValue.value}
+                                            backgroundColor={STUDY_VIEW_CONFIG.colors.theme.clinicalFilterContent}
+                                            onDelete={() => this.props.updateClinicalDataFilterByValues(chartMeta,
+                                                _.remove(clinicalDataFilter.values, value => value.value !== clinicalDataFilterValue.value))}
+                                        />
+                                    })} operation={'or'} group={false} />
```
Comment (onursumer) [inline src/pages/studyView/UserSelections.tsx]:
```suggestion
                                        })} 
                                        operation={'or'} 
                                        group={false} 
                                    />
```

**Pull request merged**